### DISCUSSION TITLE
Vimdiff layout configuration not woking

### DISCUSSION THREAD

**Discussion by sebiwi** [LazyVim/LazyVim#4480] [Q&A] [answered]
Hi, I'm trying to get the Vimdiff configuration set to `diff3_horizontal`, but when I do diffs on files with merge conflicts the layout doesn't change at all.

This is my configuration:

```
  {
    "NeogitOrg/neogit",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "sindrets/diffview.nvim",
      "nvim-telescope/telescope.nvim", -- optional
    },
    keys = {
      {
        "<LEADER>gg",
        "<cmd>Neogit<cr>",
        desc = "Neogit",
      },
      {
        "<leader>gl",
        function()
          local file = vim.fn.expand("%")
          require("neogit").action("log", "log_current", { "--", file })()
        end,
        desc = "Neogit Log for this file",
      },
      {
        "<leader>gl",
        function()
          local file = vim.fn.expand("%")
          vim.cmd([[execute "normal! \<ESC>"]])
          local line_start = vim.fn.getpos("'<")[2]
          local line_end = vim.fn.getpos("'>")[2]
          require("neogit").action("log", "log_current", { "-L" .. line_start .. "," .. line_end .. ":" .. file })()
        end,
        desc = "Neogit Log for this range",
        mode = "v",
      },
      {
        "<LEADER>gf",
        "<cmd>DiffviewFileHistory %<cr>",
        desc = "Current File History",
      },
    },
    config = true,
  },
  {
    "sindrets/diffview.nvim",
    opts = {
      default = {
        layout = "diff3_horizontal",
      },
      view = {
        file_history = {
          layout = "diff3_horizontal",
        },
        merge_tool = {
          layout = "diff3_horizontal",
        },
        diff_view = {
          layout = "diff3_horizontal",
        },
      },
    },
  },
}

```

Am I missing something?

Thanks!

**Comment by dpetka2001** [comment]
It seems that this layout is not available for `diff_view`, rather only for `merge_tool`. See relavant section in [docs](https://github.com/sindrets/diffview.nvim/blob/4516612fe98ff56ae0415a259ff6361a89419b0a/doc/diffview.txt#L508-L509).

**Comment by sebiwi** [comment]
Hey, thanks for your answer! 

According to [this](https://github.com/sindrets/diffview.nvim/tree/4516612fe98ff56ae0415a259ff6361a89419b0a?tab=readme-ov-file#merge-tool), when you open a file with conflicts, the merge tool is supposed to open instead of the diff view. I should have been more clear in my initial question (I updated it), but I’m observing this when trying to solve a merge conflict.

**Reply by dpetka2001** [reply, marked as answer]
`diff3_horizontal` is already the default one for `merge_tool`, so you shouldn't observe any change in layout. See `:h diffview-layouts` and `:h diffview.defaults`.

**Reply by sebiwi** [reply]
Ah, I see! I just realized this works when I use the `DiffviewOpen` command, but it doesn't when I use Neogit's diffview, unless I use the "diff.worktree" option.

Layer 8 issue. Thanks!
